src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/EServiceApiServiceImpl.scala

```diff
@@ -491,230 +309,120 @@ class EServiceApiServiceImpl(
         createdAt = offsetDateTimeSupplier.get(),
         activatedAt = None
       )
-      _ <- commander.ask(ref => AddCatalogItemDescriptor(current.id.toString, createdCatalogDescriptor, ref))
-    } yield createdCatalogDescriptor
-
-    onComplete(result) {
-      case Success(descriptor) => createDescriptor200(descriptor.toApi)
-      case Failure(ex)         =>
-        logger.error(s"Error while creating descriptor for e-service $eServiceId", ex)
-        createDescriptor400(problemOf(StatusCodes.BadRequest, DescriptorCreationBadRequest(eServiceId)))
-    }
+      _ <- commander(eServiceId).askWithStatus(ref =>
+        AddCatalogItemDescriptor(current.id.toString, createdCatalogDescriptor, ref)
+      )
+    } yield createdCatalogDescriptor.toApi
+
+    onComplete(result) { createDescriptorResponse[EServiceDescriptor](operationLabel)(createDescriptor200) }
   }
 
-  /** Code: 204, Message: Document deleted.
-    * Code: 404, Message: E-Service descriptor document not found, DataType: Problem
-    * Code: 400, Message: Bad request, DataType: Problem
-    */
   override def deleteEServiceDocument(eServiceId: String, descriptorId: String, documentId: String)(implicit
     toEntityMarshallerProblem: ToEntityMarshaller[Problem],
     contexts: Seq[(String, String)]
   ): Route = authorize(ADMIN_ROLE, API_ROLE) {
-    logger.info("Delete document {} of descriptor {} for e-service {}", documentId, descriptorId, eServiceId)
-    val shard: String = getShard(eServiceId)
-
-    val commander: EntityRef[Command] = getCommander(shard)
-
-    val result: Future[StatusReply[Done]] = for {
-      found         <- commander.ask(ref => GetCatalogItem(eServiceId, ref))
-      catalogItem   <- found.toFuture(EServiceNotFoundError(eServiceId))
-      document      <- extractDocument(catalogItem, descriptorId, documentId)
-      extractedFile <- extractFile(document)
```

Comment:
not sure why we were parsing the content type before deleting a file 🤔 
Are there actual use cases?

Reply:
Uhmmm...I think it comes from a previous logic.
I don't see any valid use case.
In my opinion, it can be removed.